### PR TITLE
otherlibs: Makefile with targets all, allopt, clean, partialclean

### PR DESCRIPTION
**Pull request by gasche**
I frequently get (parallel) build failures due to wrong dependencies
causing compilation failures in some otherlibs/ subdirectory. When
that happens, my workflow is to go to that subdirectory, run 'make
clean', and then compile again. But then another subdirectory is
broken for the same reason...

This new otherlibs/Makefile provides a 'clean' target that will run clean
in all subdirectories (all the ones selected to be built at
configure-time), as well as 'all', 'allopt', 'partialclean' targets
that similarly dispatch to all subdirectories.

### CODE REVIEW

**Comment by gasche** [comment]
Note: the whole-PR diff makes it look I rewrote parts of the otherlibs/Makefile file. In fact it is moved wholesale (unchanged) to otherlibs/Makefile.otherlibs.common (this Makefile is only for inclusion in from subdirectories' makefiles, and it gets in the way of otherlibs/-specific targets), and I wrote a (short) new Makefile from scratch.

**Comment by gasche** [comment]
(Anyone willing to review a short and easy PR?)

File: otherlibs/Makefile [hunk 1]
```diff
@@ -13,113 +13,23 @@
...
-.ml.cmo:
-	$(CAMLC) -c $(COMPFLAGS) $<
-
-.ml.cmx:
-	$(CAMLOPT) -c $(COMPFLAGS) $(OPTCOMPFLAGS) $<
-
-.c.$(O):
-	$(CC) -c $(OC_CFLAGS) $(OC_CPPFLAGS) $(OUTPUTOBJ)$@ $<
+# $1: target name to dispatch to all otherlibs/*/Makefile
+define dispatch_
```
Comment (alainfrisch) [inline otherlibs/Makefile:21]: Is it really worth the effort (now for the reader), just to factorize the four iterations?  I assume doubling the dollars is related to that but this is above by gmake skills.

Apart from that, LGTM.

Comment (gasche) [reply otherlibs/Makefile:21]: Worth it, probably not, but much more fun :-) I tested it to work properly, don't expect portability issues on this aspect of Make, and reused the idea from `byte_and_opt` in `tools/Makefile`.

Comment (alainfrisch) [reply otherlibs/Makefile:21]: Fair enough!

**Comment by alainfrisch** [comment]
(I've used "Rebase and Merge", hope this is ok for you...)

**Pull request merged**